src/lib/TableOfContents.svelte

```diff
@@ -0,0 +1,145 @@
+<script lang="ts">
+  import { onDestroy, onMount, setContext } from "svelte"
+  import { createPopover } from "svelte-headlessui"
+  import breakpointObserver from "$lib/breakpointObserver.js"
+  import { tableOfContents } from "../routes/store"
+  import Grid from "./components/Grid.svelte"
+  import Hamburger from "./components/Hamburger.svelte"
+
+  type Entry = {
+    title: string
+    level: number
+    id: string
+    active: boolean
+  }
+  let entries: Entry[] = []
+
+  const popover = createPopover({})
+  const size = breakpointObserver()
+  $: isSmallScreen = size.smallerThan("lg")
+  setContext("TableOfContents", {
+    addEntry: (entry: Entry) => {
+      entries.push(entry)
+      tableOfContents.set(entries)
+    },
+  })
+
+  onMount(() => {
+    tableOfContents.set(entries)
+  })
+
+  onDestroy(() => {
+    tableOfContents.set([])
+  })
+
+  function updateActiveEntry(entryId: string) {
+    entries.map((entry) => {
+      if (entry.id === entryId) {
+        entry.active = true
+      } else {
+        entry.active = false
+      }
+    })
+    tableOfContents.set(entries)
+  }
+</script>
+
+<Grid class="main-grid" container>
+  <Grid sm={$isSmallScreen ? 10 : 9}>
+    <slot />
+  </Grid>
+  {#if !$isSmallScreen}
+    <Grid sm={2} class="table-of-contents">
+      <div class="content">
+        <p>On this page</p>
+        {#each $tableOfContents as entry}
+          <a
+            on:click={updateActiveEntry(entry.id)}
```

Comment:
If a table of contents has any "active" styles it definitely should not be based on what you've clicked on last, but which section you're scrolled to. I don't think this is a useful feature in general to have and I would rather not waste time on figuring out the scroll handlers for when the items are set active

Reply:
Active styles have been removed pending the time we resolve what it should look like and how they should work